### ISSUE TITLE
Update captioned image to require contextual alternative text

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/ppa-django#322]
## Notes for testing
I've added a new required alternative text field for captioned images - this is important for accessibility, and the caption should be context-specific, so I've made it specific to the image as included in a page rather than attached to the image generically.

This change does unfortunately mean that editing existing content with images will require that you add alternative text. If you think this is too much of a problem, let me know; we could make it optional (but should be clear to editors that it's highly recommended to use).

**Comment by mnaydan** [comment]
I tested this on an SVG and a regular image and it's fine. I don't think editing existing content will be too much of a problem (Gissoo's essay would be the biggest bear but doable). 

For the SVGs, I don't understand the difference between the alternative text and the extended description. It seems redundant to me. I would imagine I'd copy and paste the same text in both fields. Is the alternative text supposed to be markedly shorter? Do we have the extended description field only on SVGs because SVGs will likely require more extensive descriptions as data visualizations?

**Comment by rlskoeser** [comment]
@mnaydan the extended description should be optional - please test to confirm that it is optional, if you did not already. 

The extended description is meant to make SVGs of data visualizations more accessible; alternate text needs to be short (screen readers don't treat it like regular text, so you can't pause or skip around). The typical solution to make a visualization accessible for vision-impaired users is to provide a chart with the numbers, but that isn't really the same thing and it doesn't work for large datasets; this is a different approach, giving a place to summarize the insights from the visualization.

**Comment by mnaydan** [comment]
@rlskoeser Thanks for explaining the difference. The alternate text field is required and trying to publish without entering text into the field rightly generates a validation error. The extended description is optional and the form can be submitted with or without text in that field. Since it's doing what we want it to do I'll close out the ticket.
